Sources/BraveWallet/Crypto/Stores/NFTDetailStore.swift

```diff
@@ -74,7 +79,9 @@ class NFTDetailStore: ObservableObject {
       
       if nftMetadata == nil {
         isLoading = true
-        nftMetadata = await rpcService.fetchNFTMetadata(for: nft)
+        if let newNftMetadata = await rpcService.fetchNFTMetadata(for: nft, ipfsApi: self.ipfsApi) {
+          nftMetadata = newNftMetadata
+        }
```

Comment:
nit: this could be left as one line:
```suggestion
        nftMetadata = await rpcService.fetchNFTMetadata(for: nft, ipfsApi: self.ipfsApi)
```